src/github_type.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2023 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""Type of returned data from GitHub web API."""
+
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import List, TypedDict
+
+from typing_extensions import NotRequired
+
+
+class GitHubRunnerStatus(Enum):
+    """Status of runner on GitHub."""
+
+    ONLINE = "online"
+    OFFLINE = "offline"
+
+
+class RunnerApplication(TypedDict, total=False):
+    """Information on a single runner application."""
+
+    os: str
+    architecture: str
+    download_url: str
+    filename: str
+    temp_download_token: NotRequired[str]
+    sha256_checksum: NotRequired[str]
+
+
+RunnerApplicationList = List[RunnerApplication]
+
+
+class SelfHostedRunnerLabel(TypedDict, total=False):
+    """A single label of self-hosted runners."""
+
+    id: NotRequired[int]
+    name: str
+    type: NotRequired[str]
+
+
+class SelfHostedRunner(TypedDict):
+    """Information on a single self-hosted runner."""
+
+    id: int
+    name: str
+    os: str
+    status: GitHubRunnerStatus
+    busy: bool
+    labels: list[SelfHostedRunnerLabel]
+
+
+class SelfHostedRunnerList(TypedDict):
+    """Information on a collection of self-hosted runners."""
```

Comment:
These seem to be missing attribute descriptions

Reply:
Minor comments